### PR TITLE
[viewer] feat: Paid Post Seo support

### PR DESCRIPTION
**Pull request by maimonav**

### CODE REVIEW

File: packages/viewer/web/src/utils/paywallSeo.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const getPaywallSeoClass = ({ className = 'paywall', size = 3 }, blockIndex) =>
```
Comment (shaulgo) [inline packages/viewer/web/src/utils/paywallSeo.js:1]:
```suggestion
export const getPaywallSeoClass = ({ className = 'paywall', index = 3 }, blockIndex) =>
```

File: packages/viewer/web/src/utils/convertContentState.jsx [hunk 2]
```diff
@@ -173,9 +176,9 @@ const getEntities = (typeMappers, pluginProps, styles, addAnchorsPrefix) => {
 
 const normalizeContentState = contentState => ({
   ...contentState,
-  blocks: contentState.blocks.map(block => {
+  blocks: contentState.blocks.map((block, blockIndex) => {
```
Comment (shaulgo) [inline packages/viewer/web/src/utils/convertContentState.jsx:179]: We should be able to get the blockIndex from content state and blockKey. BlockKey was already added as id to every block. 

You can add a function like this `contentState.blocks.findIndex(block => block.key === blockProps.keys[childIndex])` to utils. And use it to get the index.

File: packages/viewer/web/src/utils/paywallSeo.js (hunk 1 above)
Comment (maimonav) [reply packages/viewer/web/src/utils/paywallSeo.js:1]: done.

File: packages/viewer/web/src/utils/convertContentState.jsx (hunk 2 above)
Comment (maimonav) [reply packages/viewer/web/src/utils/convertContentState.jsx]: Done.

File: packages/common/web/src/Utils/draftUtils.js [hunk 3]
```diff
@@ -14,3 +14,6 @@ export const getLinkRangesInBlock = (block, contentState) => {
   });
   return ranges;
 };
+
+export const getBlockIndex = (contentState, blockKey) =>
```
[the remark this answers is not in the document]
Comment (maimonav) [reply packages/common/web/src/Utils/draftUtils.js:18]: Done.

File: packages/viewer/web/src/utils/convertContentState.jsx [hunk 4]
```diff
@@ -98,7 +100,9 @@ const getBlocks = (contentState, mergedStyles, textDirection, context, addAnchor
                 mergedStyles[style]
               ),
               depthClassName(depth),
-              directionClassName
+              directionClassName,
+              isPaywallSeo(context.seoMode) &&
```
Comment (shaulgo) [inline packages/viewer/web/src/utils/convertContentState.jsx:104]: isPaywallSeo(context.seoMode, blockIndex) && context.seoMode.paywall.className

File: packages/viewer/web/src/getPluginViewers.jsx [hunk 5]
```diff
@@ -145,14 +155,15 @@ PluginViewer.propTypes = {
...
+  blockIndex: PropTypes.number,
 };
 
 PluginViewer.defaultProps = {
   styles: {},
 };
 
 //return a list of types with a function that wraps the viewer
-const getPluginViewers = (typeMappers, context, styles, addAnchorFnc) => {
+const getPluginViewers = (contentState, typeMappers, context, styles, addAnchorFnc) => {
```
Comment (shaulgo) [inline packages/viewer/web/src/getPluginViewers.jsx:166]: you can check if you prefer to add it to the context

Comment (maimonav) [reply packages/viewer/web/src/getPluginViewers.jsx:166]: Yeah I agree I think it's better, I'll avoid passing it in multiple functions and simply use `context.contentState.`
I'll add it tomorrow and merge.

**Pull request merged**